test.sh

```diff
@@ -0,0 +1,178 @@
+
+# configure Nginx on macOS
+CreateaMac() {
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        
+        sudo tee $CONFIG_FILE_MAC > /dev/null <<EOL
+worker_processes 1;
+events { worker_connections 1024; }
+http {
+    server {
+        listen $DEFAULT_PORT;
+        server_name localhost;
+        root /usr/local/var/www;
+        index index.html;
+    }
+EOL
+
+        [ -n "$EXTRA_PORT" ] && sudo tee -a $CONFIG_FILE_MAC > /dev/null <<EOL
+    server {
+        listen $EXTRA_PORT;
+        server_name localhost;
+        root /usr/local/var/www;
+        index index.html;
+    }
+EOL
+
+        sudo tee -a $CONFIG_FILE_MAC > /dev/null <<EOL
+}
+EOL
+
+        # Restart brew Nginx
+        brew services restart nginx
+    fi
+}
+
+# configure Nginx on Linux
+CreateaLinux() {
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+       
+        sudo tee $CONFIG_FILE_LINUX > /dev/null <<EOL
+worker_processes 1;
+events { worker_connections 1024; }
+http {
+    server {
+        listen $DEFAULT_PORT;
+        server_name localhost;
+        root /var/www/html;
+        index index.html;
+    }
+EOL
+
+        [ -n "$EXTRA_PORT" ] && sudo tee -a $CONFIG_FILE_LINUX > /dev/null <<EOL
+    server {
+        listen $EXTRA_PORT;
+        server_name localhost;
+        root /var/www/html;
+        index index.html;
+    }
+EOL
+
+        sudo tee -a $CONFIG_FILE_LINUX > /dev/null <<EOL
+}
+EOL
+
+        # Reiniciar Nginx
+        sudo systemctl restart nginx
+    fi
+}
+
+# clean if a subfunction "clean" is called
+clean_up() {
+            #linux
+    if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+        # Stop and uninstall Nginx
+        sudo systemctl stop nginx
+        if [[ -f /etc/debian_version ]]; then
+            sudo apt-get remove --purge -y nginx
+            sudo apt-get autoremove -y
+        elif [[ -f /etc/redhat-release ]]; then
+            sudo yum remove -y nginx
+        elif [[ -f /etc/arch-release ]]; then
+            sudo pacman -R --noconfirm nginx
+        else
+           cho "Distribution of Linux not soported"
+            exit 1
+        fi
+        # remove configuration files and directories linux
+        sudo rm -rf /etc/nginx /var/www/html /var/log/nginx
+
+    #mac
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        # STOP and uninstall Nginx
+        brew services stop nginx
+        brew uninstall nginx
+        #  remove configuration files and directories mac
+        sudo rm -rf /usr/local/etc/nginx /usr/local/var/www /usr/local/var/log/nginx
+    else
+        echo "Sistema operativo no soportado"
+        exit 1
+    fi
+    echo "Limpieza completada. Nginx y todos los archivos relacionados han sido eliminados."
+}
+
+# Lógica principal
+if [[ "$1" == "clean" ]]; then
+    clean_up
+    exit 0
+fi
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    InstallNL
+    CreateaLinux
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    InstallMAC
+    CreateaMac
+else
+    echo "Sistema operativo no soportado"
+    exit 1
+fi
```

Comment:
Would be better to wrap this up into a function in order to keep it clean and consistent with the function base style you're using, something such as:
```suggestion
main() {
    # Lógica principal
    if [[ "$1" == "clean" ]]; then
        clean_up
        exit 0
    fi
    
    case "$OSTYPE" in
        "linux-gnu"*)
            install_nginx_linux
            ;;
        "darwin"*)
            install_nginx_macos
            ;;
        *)
            echo "Sistema operativo no soportado"
            exit 1
            ;;
    esac
    open_browser
}

main
```